app/api/delete_indices_endpoint.py

```diff
@@ -0,0 +1,36 @@
+from fastapi import APIRouter, Depends, HTTPException
+from app.config import server
+from tracardi.config import NAME
+from tracardi.service.storage.elastic_client import ElasticClient
+from elasticsearch import ElasticsearchException
+from .auth.permissions import Permissions
+from typing import Optional
+from hashlib import md5
+
+router = APIRouter(
+    dependencies=[Depends(Permissions(roles=["admin"]))]
+)
+
+
+@router.delete("/indices/version/{version}", tags=["index"], include_in_schema=server.expose_gui_api)
+async def delete_old_indices(version: str, codename: Optional[str] = None):
+
+    if codename is None:
+        codename = md5(version.encode('utf-8')).hexdigest()[:5]
+
+    if codename == NAME:
+        raise HTTPException(status_code=409, detail="Cannot delete currently connected indices.")
+
+    try:
+        es = ElasticClient.instance()
+        indices = await es.list_aliases()
+        to_delete = [index for index in indices if index.startswith(f"{version.replace('.', '')}.{codename}.tracardi-")]
```

Comment:
This logic is probably somewhere else as well. If we decide to change it is must be in one place. Maybe index should have it.

Reply:
Please use objects e.g

```python
def is_version(version: Version):
  ...
```

You should have it if you make it out of input parameters.